### PR TITLE
Enable invalid credential handling for LXD

### PR DESCRIPTION
**Pull request by Veebers**
## Description of change

Invalidate the credential whenever we get an authorisation error back from any provider action.

### CODE REVIEW

File: provider/lxd/environ_broker_test.go [hunk 1]
```diff
@@ -350,7 +361,28 @@ func (s *environBrokerSuite) TestStartInstanceNoTools(c *gc.C) {
...
+	)
+
+	env := s.NewEnviron(c, svr, nil)
+	_, err := env.StartInstance(s.callCtx, s.GetStartInstanceArgs(c, "bionic"))
+	c.Assert(err, gc.ErrorMatches, "not authorized")
+	c.Assert(s.invalidCredential, jc.IsTrue)
+}
+
 func (s *environBrokerSuite) TestStopInstances(c *gc.C) {
+	c.Assert(s.invalidCredential, jc.IsFalse)
```
Comment (anastasiamac) [inline provider/lxd/environ_broker_test.go]: Why is this here?
If we are checking that the credential is valid in the beginning of the test, I would have expected that the test makes some calls to invalidate the credential and then check that it is indeed invalid at the end. But it does not seem to be the case here :)

I think you wanted to make this call int he beginning of TestStopInstancesInvalidCredentials... It is missing from there :)

File: provider/lxd/environ_test.go [hunk 2]
```diff
@@ -200,6 +250,121 @@ func (s *environSuite) TestDestroyController(c *gc.C) {
...
+
+	s.Client.Containers = append(s.Client.Containers, *machine0)
+
+	// RemoveContainers will error not-auth.
+	s.Client.Stub.SetErrors(nil, nil, nil, nil, nil, errTestUnAuth)
+
+	err := s.Env.DestroyController(s.callCtx, s.Config.UUID())
+	c.Assert(err, gc.ErrorMatches, "not authorized")
+	c.Assert(s.invalidCredential, jc.IsTrue)
+	s.Stub.CheckCalls(c, []gitjujutesting.StubCall{
```
Comment (anastasiamac) [inline provider/lxd/environ_test.go]: I would have expected to see a check that controller was not destroyed here :)

**Comment by anastasiamac** [comment]
[adjust lxd provider to use callback to invalidate cloud credential when needed](https://trello.com/c/ymHuqezY/362-adjust-lxd-provider-to-use-callback-to-invalidate-cloud-credential-when-needed)

**Comment by Veebers** [comment]
$$merge$$
